pkg/k8s/version/version.go

```diff
@@ -137,21 +162,94 @@ func fallbackDiscovery(client kubernetes.Interface) error {
 		return nil
 	}
 
-	switch t := err.(type) {
-	case *errors.StatusError:
-		if t.ErrStatus.Code == http.StatusNotFound {
-			log.WithError(err).Info("Unable to retrieve EndpointSlices for default/kubernetes. Disabling EndpointSlices")
-			// StatusNotFound is a safe error, EndpointSlices are
-			// disabled and the agent can continue
-			return nil
-		}
+	if errors.IsNotFound(err) {
+		log.WithError(err).Info("Unable to retrieve EndpointSlices for default/kubernetes. Disabling EndpointSlices")
+		// StatusNotFound is a safe error, EndpointSlices are
+		// disabled and the agent can continue.
+		return nil
 	}
 
 	// Unknown error, we can't derive whether to enable or disable
-	// EndpointSlices and need to error out
+	// EndpointSlices and need to error out.
 	return fmt.Errorf("unable to validate EndpointSlices support: %s", err)
 }
 
+func leasesFallbackDiscovery(client kubernetes.Interface, conf k8sconfig.Configuration) error {
+	// K8sEnableLeasesFallbackDiscovery is used to fallback leases discovery to directly
+	// probing the API when we cannot discover API groups.
+	// We require to check for Leases capabilities in operator only, which uses Leases
+	// for leader election purposes in HA mode.
+	if !conf.K8sLeasesFallbackDiscoveryEnabled() {
+		log.Debugf("Skipping Leases support fallback discovery")
+		return nil
+	}
+
+	cached.mutex.RLock()
+	// Here we check if cilium has required authorization to check for leases object or not.
+	// Prior to v1.14 leases.coordination.k8s.io/v1 was not supported and hence cilium
```

Comment:
I see a reference to document the version so that's good :+1: , but it would be nice to centralize it.